### PR TITLE
chore!: Rename OriginalSquareSize to SquareSize so that same name to same thing.

### PR DESCRIPTION
**Pull request by HoytRen**
## Description

Since this is my first PR to Celestia. I list the steps for your reference.

1. fork the https://github.com/celestiaorg/celestia-core
2. create the branch according to issue
3. build and thest the unchanged code ( here are 2 errors of test, but I don't think it's relative, one is about http, and other is state index )
4. work out a solution ( include checking potencial affects to downstream projects like celestia-app, it will be break if we don't change it )
5. modify codes
6. review codes by compare to orign ( here is a unused function in block.go, but I believe I'm not relative )
7. build and test the code ( here are still the 2 errors )
8. compare to orign again to ensure the auto-generate file works
9. submit and push to my fork
10. create PR here

Closes: #853

### CODE REVIEW

File: proto/tendermint/types/types.pb.go [hunk 1]
```diff
@@ -422,10 +422,10 @@ type Data struct {
...
 	Txs [][]byte `protobuf:"bytes,1,rep,name=txs,proto3" json:"txs,omitempty"`
 	// field number 2 is reserved for intermediate state roots
-	Evidence           EvidenceList `protobuf:"bytes,3,opt,name=evidence,proto3" json:"evidence"`
-	Messages           Messages     `protobuf:"bytes,4,opt,name=messages,proto3" json:"messages"`
-	OriginalSquareSize uint64       `protobuf:"varint,5,opt,name=original_square_size,json=originalSquareSize,proto3" json:"original_square_size,omitempty"`
-	Hash               []byte       `protobuf:"bytes,6,opt,name=hash,proto3" json:"hash,omitempty"`
+	Evidence   EvidenceList `protobuf:"bytes,3,opt,name=evidence,proto3" json:"evidence"`
+	Messages   Messages     `protobuf:"bytes,4,opt,name=messages,proto3" json:"messages"`
+	SquareSize uint64       `protobuf:"varint,5,opt,name=original_square_size,json=squareSize,proto3" json:"square_size,omitempty"`
+	Hash       []byte       `protobuf:"bytes,6,opt,name=hash,proto3" json:"hash,omitempty"`
```
Comment (evan-forbes) [inline proto/tendermint/types/types.pb.go:428]: we should never be editing these files, instead, we need to change the proto file and then regenerate the bindings using `make proto-gen`

**Review by evan-forbes — CHANGES_REQUESTED** [review]
we also need to change the proto files as well, and then regenerate them. The proto files should never be edited by hand :slightly_smiling_face:

**Comment by rootulp** [comment]
Thanks for the detailed steps in the PR Description @HoytRen 
I hope https://github.com/celestiaorg/celestia-core/pull/856 makes the steps in the README clearer for future contributors

**Comment by HoytRen** [comment]
I updated the PR. It seems here need 1 more check step after bulid to ensure the auto-generated file works, so I updated the discription, then @rootulp could use it later.

**Comment by HoytRen** [comment]
When I buld the project. Sometimes I got a dead lock error. I fell it could be important, then I put it here ( I'am not sure if I shoud create a new issue or I should look into it first ).
[error01.txt](https://github.com/celestiaorg/celestia-core/files/9571595/error01.txt)

**Comment by HoytRen** [comment]
By the way, I'm thinking about that could we remove these auto-generated files from git ?

File: proto/tendermint/types/types.proto [hunk 2]
```diff
@@ -91,7 +91,7 @@ message Data {
   // field number 2 is reserved for intermediate state roots
   EvidenceList           evidence                 = 3 [(gogoproto.nullable) = false];
   Messages               messages                 = 4 [(gogoproto.nullable) = false];
-  uint64                 original_square_size     = 5;
+  uint64                 square_size     = 5;
```
Comment (evan-forbes) [inline proto/tendermint/types/types.proto:94]: we also need to format this, which can be done using `make proto-format`

**Review by evan-forbes — CHANGES_REQUESTED** [review]

**Comment by evan-forbes** [comment]
> By the way, I'm thinking about that could we remove these auto-generated files from git ?

I don't think that's something we can safely do. if we removed them from git, then people would have to generate them before building, which is cumbersome. More importantly tho, depending on people's setup, the compilation of these files can differ (this one reason why the cosmos-sdk uses a docker image to compile), which would likely lead to a break in consensus. Also, this is a significant departure from what upstream tendermint does, which is a non-starter.

**Comment by HoytRen** [comment]
Yes, its a sound reason for keeping these files.

Thank you very much for telling me the make target, then I don't need to format it manually. I should not have a mistake like this.

I fixed it and please review it again. Feel free to tell me if here are something could be improved. I'm very happy to learn all of these.

**Review by evan-forbes — CHANGES_REQUESTED** [review]
@HoytRen one of the CI failures is expected, but the one that checks that the proto-files were generated correctly should not. It looks like you need to simply regenerate the proto files again (`make proto-gen`) as the formatting caused some of the docs to change, which need to be reflected in the generated files as well. Then we should be good to merge!!

**Comment by HoytRen** [comment]
It seems OK this time. thank you very much for your encourage.

I add proto-format and proto-gen to my local make file, let them always run, so that I don't repeat these mistaks.

Please review again.

**Review by evan-forbes — APPROVED** [review]
good work @HoytRen, thanks again

will wait to merge until @rootulp has a chance to double check my review

**Comment by rootulp** [comment]
Thanks a ton for this contribution @HoytRen !

**Comment by HoytRen** [comment]
Really happy to complete the task with the help from both of you 2. Let me think about next :)

**Pull request merged**